### PR TITLE
RN-376: Updated database dump script

### PR DESCRIPTION
**Pull request by rohan-bes**
### Issue RN-376:

Few small improvements to the database dump script post the RDS changes.

### Changes:
- No longer requires password input
- Uses compression
- Shows loading spinner to indicate progress

### CODE REVIEW

File: packages/database/scripts/dumpDatabase.sh [hunk 1]
```diff
@@ -11,6 +11,26 @@ Options:
 EOF
 }
 
+# https://stackoverflow.com/questions/12498304/using-bash-to-display-a-progress-indicator
+function show_loading_spinner() {
```
Comment (rohan-bes) [inline packages/database/scripts/dumpDatabase.sh:15]: Probably unnecessary and just having a little bit of fun. But it wasn't tricky and does make it clearer that something is actually happening as we wait the 30-40 mins it takes to dump the database

File: packages/database/scripts/dumpDatabase.sh [hunk 2]
```diff
@@ -55,7 +75,11 @@ target_path="$(
     cd "$target_dir"
     pwd
 )/$DUMP_FILE_NAME"
+target_zip_path="$target_path.gz"
+
+show_loading_spinner "Dumping database to $target_zip_path" "PGPASSWORD=$DB_PG_PASSWORD pg_dump \"host=$host user=postgres dbname=tupaia sslmode=require sslkey=$identity_file\" -Z1 -f $target_zip_path"
```
Comment (rohan-bes) [inline packages/database/scripts/dumpDatabase.sh:80]: Adding the `-Z1` argument turns on compression of the dump file. I skipped doing this previously as it didn't really seem to speed up the process at all.

However for the folks at home saving 8GB of data download each time you dump the database is probably worth it

**Review by edmofro — APPROVED** [review]
lgtm, thanks for this!

I'm thinking of this as a helpful stopgap until we get something better in place, being either:
- A smaller db for local dev, either through generation of dummy data or stripping the live db; or
- Pointing to an RDS environment for local dev

**Comment by rohan-bes** [comment]
Thanks for the review @edmofro 

Agreed about this being a stop gap for now. Hopefully after moving some of our larger projects to the data lake we won't have quite as large a database, but it's still best if we move away from devs having real data in their local environments...

**Comment by rohan-bes** [comment]
@edmofro added a check for the postgres user env vars, as they probably won't be there in most people's local .env file

**Pull request merged**